packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts

```diff
@@ -153,33 +165,35 @@ const updateInventoryAndReservations = async (
 ) => {
   const { inventoryService, locationId } = context
 
-  fulfillments.map(async ({ items }) => {
-    await inventoryService.validateInventoryAtLocation(
-      items.map(({ item, quantity }) => ({ ...item, quantity } as LineItem)),
-      locationId
-    )
-
-    await Promise.all(
-      items.map(async ({ item, quantity }) => {
-        if (!item.variant_id) {
-          return
-        }
+  await Promise.all(
+    fulfillments.map(async ({ items }) => {
+      await inventoryService.validateInventoryAtLocation(
```

Comment:
todo: we should use the transaction (related to the previous comment) and create a `const inventoryServiceTx`

Reply:
same response as above comment 😉